Build failure: libloot

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

build a package with libloot dependency, in my case it's Limo

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317866256

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/24s2wz2wax1xp1h7nq6j426kan5kmdmc-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/52dfc4abwbiblip9a4sf8lsypaqk05vy-deps.patch
patching file CMakeLists.txt
Hunk #1 succeeded at 81 with fuzz 2 (offset 2 lines).
Hunk #2 succeeded at 156 (offset 2 lines).
Hunk #3 succeeded at 310 (offset 1 line).
patching file cmake/tests.cmake
Hunk #1 succeeded at 110 (offset 2 lines).
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO>
-- The C compiler identification is GNU 15.2.0
-- The CXX compiler identification is GNU 15.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/g++ - skipp>
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find Git (missing: GIT_EXECUTABLE) 
-- Git revision: unknown
-- Found ICU: /nix/store/55fckh05snmnaynvf505apr655y68nzc-icu4c-76.1-dev/include (found version "76.1") found com>
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
CMake Warning at /nix/store/wmb16s8hw1vd25vvidwnncpzq126lcz3-cmake-4.1.2/share/cmake-4.1/Modules/FetchContent.cma>
  Relative source directory specified.  This is not safe, as it depends on
  the calling directory scope.

    FETCHCONTENT_SOURCE_DIR_TESTING-PLUGINS --> ../testing-plugins
Call Stack (most recent call first):
  /nix/store/wmb16s8hw1vd25vvidwnncpzq126lcz3-cmake-4.1.2/share/cmake-4.1/Modules/FetchContent.cmake:2384 (__Fetc>
  cmake/tests.cmake:27 (FetchContent_MakeAvailable)
  CMakeLists.txt:416 (include)


-- Configuring done (0.5s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j4 SHELL=/nix/store/lw117lsr8d585xs63kx5k233impyrq7q-bash-5.3p3/bin/bash
[  2%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/binary.cpp.o
[  2%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/convert.cpp.o
[  3%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/depthguard.cpp.o
[  4%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/directives.cpp.o
[  5%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emit.cpp.o
[  6%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitfromevents.cpp.o
[  7%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitter.cpp.o
[  8%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitterstate.cpp.o
[  9%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/emitterutils.cpp.o
[ 10%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/exceptions.cpp.o
[ 11%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/exp.cpp.o
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:221:11: error: 'uint16_t' was not declare>
  221 | std::pair<uint16_t, uint16_t> EncodeUTF16SurrogatePair(int codePoint) {
      |           ^~~~~~~~
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:13:1: note: 'uint16_t' is defined in head>
   12 | #include "yaml-cpp/null.h"
  +++ |+#include <cstdint>
   13 | #include "yaml-cpp/ostream_wrapper.h"
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:221:21: error: 'uint16_t' was not declare>
  221 | std::pair<uint16_t, uint16_t> EncodeUTF16SurrogatePair(int codePoint) {
      |                     ^~~~~~~~
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:221:21: note: 'uint16_t' is defined in he>
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:221:29: error: template argument 1 is inv>
  221 | std::pair<uint16_t, uint16_t> EncodeUTF16SurrogatePair(int codePoint) {
      |                             ^
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:221:29: error: template argument 2 is inv>
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp: In function 'int YAML::Utils::{anonymous>
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:222:9: error: 'uint32_t' does not name a >
  222 |   const uint32_t leadOffset = 0xD800 - (0x10000 >> 10);
      |         ^~~~~~~~
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:222:9: note: 'uint32_t' is defined in hea>
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:225:5: error: 'leadOffset' was not declar>
  225 |     leadOffset | (codePoint >> 10),
      |     ^~~~~~~~~~
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:227:3: error: cannot convert '<brace-encl>
  227 |   };
      |   ^
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp: In function 'void YAML::Utils::{anonymou>
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:246:55: error: request for member 'first'>
  246 |     WriteDoubleQuoteEscapeSequence(out, surrogatePair.first, stringEscapingStyle);
      |                                                       ^~~~~
/nix/store/2h8pksh0lm6rspxbw51f0bp8p5sfh5mf-source/src/emitterutils.cpp:247:55: error: request for member 'second>
  247 |     WriteDoubleQuoteEscapeSequence(out, surrogatePair.second, stringEscapingStyle);
      |                                                       ^~~~~~
[ 12%] Building CXX object _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/src/memory.cpp.o
make[2]: *** [_deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/build.make:191: _deps/yaml-cpp-build/CMakeFiles/yaml-c>
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:211: _deps/yaml-cpp-build/CMakeFiles/yaml-cpp.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 25.11 (Xantusia), 25.11.20251229.89dbf01`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/rv4z0vd7v0pkadinxd1c1xmd4b850ldr-source`

### Notify maintainers

@TomaSajt 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/li/libloot/package.nix
+++ b/pkgs/by-name/li/libloot/package.nix
@@ -119,7 +119,13 @@ stdenv.mkDerivation (finalAttrs: {
       owner = "loot";
       repo = "yaml-cpp";
       tag = "0.8.0+merge-key-support.2";
-      hash = "sha256-whYorebrLiDeO75LC2SMUX/8OD528BR0+DEgnJxxpoQ=";
+
+      # fixes error: 'uint16_t' was not declared in this scope
+      postFetch = ''
+        sed -e '1i #include <cstdint>' -i "$out/src/emitterutils.cpp"
+      '';
+
+      hash = "sha256-5xbqOI4L3XCqx+4k6IcZUwOdHAfbBy7nZgRKGkRJabQ=";
     };
 
     buildRustFFIPackage =

